contrib/src/main/scala/akka/stream/contrib/FocusedFlow.scala

```diff
@@ -0,0 +1,241 @@
+package akka.stream.contrib
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream._
+import akka.stream.contrib.FocusedSource.IncompleteFocusedSourceShape
+import akka.stream.scaladsl.{BidiFlow, Flow, FlowOpsMat, GraphDSL, Keep, RunnableGraph, Sink, Source}
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+
+import scala.annotation.unchecked.uncheckedVariance
+import scala.collection.immutable
+
+object FocusedFlow {
+  implicit class FocusedFlowOps[Tag, Src, In, Mat](from: Flow[Src, (Tag, In), Mat]) {
+    def focus = new FocusedFlow(BidiFlow.fromFlowsMat(from, Flow[In])(Keep.left))
+  }
+
+  def main(args: Array[String]): Unit = {
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorMaterializer()
+
+    val flow: Flow[(Int, String), (Int, String), NotUsed] = Flow[(Int, String)]
+      .focus
+        .map(_.toUpperCase)
+        .filterNot(_.startsWith("T"))
```

Comment:
Hmm, interesting question. I guess that would depend on your definition of 'work', but most probably not in the way you (or the user) intends.

For mapAsync, assuming parallelism > 1 it won't work as expected. For mapConcat, it could work, assuming it doesn't buffer internally.